src/core/qgsfileutils.h

```diff
@@ -98,8 +100,60 @@ class CORE_EXPORT QgsFileUtils
      * \returns List of strings of the first matching path in unix format.
      * \since QGIS 3.12
      */
-    static QStringList findFile( const QString &file, const QString &basepath = QString(), int maxClimbs = 4, int searchCeiling = 4, const QString &currentDir = QString() );
+    static QStringList findFile( const QString &file, const QString &basepath = QString(), int maxClimbs = 4, int searchCeiling = 4, const QString &currentDir = QString(), QgsFileSearchTask *task = nullptr );
```

Comment:
instead of passing the task as an argument here, the correct approach is to pass an optional "QgsFeedback" object to functions like this. Then in your task you create a new QgsFeedback object for passing to this function, and proxy the QgsFeedback objects' cancelation/progress reporting to the task via connections.

It keeps utility functions like findFile nice and generic and available for in a wider variety of contexts (e.g. we could easily then add a processing algorithm for finding files... which... actually sounds like a great idea :laughing: )

Reply:
Done, I wasn't sure if the feedback was mostly for processing or had any other dependencies as I wanted something linked to the dialog if possible.